src/assets.rs

```diff
@@ -226,33 +222,71 @@ impl<'a> DerefMut for AssetEditor<'a> {
 
 /// Assets stored in an assets store.
 pub struct Assets {
+    /// A key-value store for assets.
     pub store: AssetsStore,
+
+    /// A set of asset codes loaded from verified asset libraries.
+    pub verified_assets: HashSet<AssetCode>,
 }
 
 impl Assets {
     #![allow(dead_code)]
 
-    /// Create assets.
+    /// Create unverified assets.
+    ///
+    /// To load verified assets afterwards, use `verify_asset`.
     pub fn new<L: Ledger>(store: AssetsStore) -> Result<Self, KeystoreError<L>> {
-        Ok(Self { store })
+        Ok(Self {
+            store,
+            verified_assets: HashSet::new(),
+        })
     }
 
     /// Iterate through the assets.
     pub fn iter(&self) -> impl Iterator<Item = Asset> + '_ {
         self.store.iter().cloned()
     }
 
-    /// Get the asset by the code.
+    /// Load a verified asset library with its trusted signer.
+    ///
+    /// Call this function at least once after assets resource is created.
+    pub async fn verify_assets<L: Ledger>(
+        &mut self,
+        trusted_signer: &VerKey,
+        library: VerifiedAssetLibrary,
+    ) -> Result<(), KeystoreError<L>> {
+        if let Some(assets) = library.open(trusted_signer) {
+            for asset in &assets {
+                self.verified_assets.insert(asset.definition.code);
+            }
+            Ok(())
+        } else {
+            Err(KeystoreError::AssetVerificationError)
+        }
+    }
+
+    /// Get the asset by the code from the store.
+    ///
+    /// Sets the asset as verified if it's in the verified set.
```

Comment:
I don't think this should be a doc comment. The fact that we keep a verified set in memory and set the `verified` flag of each asset on load is an implementation detail. Semantics-wise, this function doesn't "set" anything -- it's just a getter.

Probably we should have some module-level or struct-level documentation explaining the concept of verified assets and how/why verified status is not persisted. That would be the place to add an implementation note about the in-memory set and the set-on-load behavior.

Reply:
Modified and moved the comment. Does it (and other similar places) look better?